src/components/compliance-form.tsx

```diff
@@ -0,0 +1,686 @@
+import { zodResolver } from "@hookform/resolvers/zod";
+import { AlertCircle, ExternalLink } from "lucide-react";
+import { useCallback, useEffect, useMemo, useRef, useState } from "react";
+import { useForm } from "react-hook-form";
+import { toast } from "sonner";
+import { ComplianceStatus, type StatusType } from "./compliance-status";
+
+type ComplianceResponse = {
+  agreementUrl: string | null;
+  kycUrl: string | null;
+  status: {
+    agreementStatus: StatusType;
+    kycStatus: StatusType;
+    isCompliant: boolean;
+  };
+};
+
+export function ComplianceForm({ user }: { user: User }) {
+  const [complianceData, setComplianceData] =
+    useState<ComplianceResponse | null>(null);
+  const [showAgreementModal, setShowAgreementModal] = useState(false);
+
+  const iframeRef = useRef<HTMLIFrameElement>(null);
+  const TRUSTED_ORIGINS = useMemo(() => {
+    const origins = process.env.NEXT_PUBLIC_TRUSTED_ORIGINS
+      ? process.env.NEXT_PUBLIC_TRUSTED_ORIGINS.split(",")
+      : ["https://request.network", "https://core-api-staging.pay.so"];
+
+    // Add localhost in development
+    if (process.env.NODE_ENV === "development") {
+      origins.push("http://localhost:3000");
+    }
+
+    return origins;
+  }, []);
+
+  // Fetch compliance status when component mounts
+  const { isLoading: isLoadingStatus, refetch: getComplianceStatus } =
+    api.compliance.getComplianceStatus.useQuery(
+      { clientUserId: user?.email ?? "" },
+      {
+        // Only fetch if we have a user email
+        enabled: !!user?.email,
+        // Don't refetch on window focus to avoid disrupting the user
+        refetchOnWindowFocus: false,
+        onSuccess: (data) => {
+          // If the user is already compliant, we can skip the form
+          if (data.success) {
+            // Set compliance data with status from the API
+            setComplianceData({
+              agreementUrl: data.data.agreementUrl ?? null,
+              kycUrl: data.data.kycUrl ?? null,
+              status: {
+                agreementStatus: data.data.agreementStatus as StatusType,
+                kycStatus: data.data.kycStatus as StatusType,
+                isCompliant: data.data.isCompliant,
+              },
+            });
+          }
+        },
+      },
+    );
+
+  const submitComplianceMutation =
+    api.compliance.submitComplianceInfo.useMutation({
+      onSuccess: (response) => {
+        if (response.success) {
+          setComplianceData(response.data);
+          getComplianceStatus();
+          toast.success("Compliance information submitted successfully");
+        } else {
+          toast.error(
+            response.message || "Failed to submit compliance information",
+          );
+        }
+      },
+      onError: (error) => {
+        console.error("Compliance submission error:", error);
+        toast.error(
+          error.message ||
+            "Failed to submit compliance information. Please try again.",
```

Comment:
Consider combining the or conditions together

```suggestion
          "Failed to submit compliance information. Error: {error.message}",
```